### ISSUE TITLE
Docker won't start after fresh install

### ISSUE THREAD

**Issue by khards** [docker/for-mac#1449]
### Information

  - Full output of the diagnostics from "Diagnose & Feedback" in the menu
  - A reproducible case if this is a bug, Dockerfiles FTW
  - Page URL if this is a docs issue or the name of a man page

### Steps to reproduce the behavior

  1. ...Install Docker
  2. ...It throws an error

Docker for Mac: version: 17.03.0-ce-mac2 (1d7d97bbb)
macOS: version 10.12.3 (build: 16D32)
logs: /tmp/D4E400B8-ED16-40F6-A513-834C367931B6/20170320-115252.tar.gz
failure: com.docker.slirp is not running
[OK]     vmnetd
[OK]     dns
[ERROR]  driver.amd64-linux
         com.docker.driver.amd64-linux -db is not running
[OK]     virtualization VT-X
[OK]     app
[ERROR]  moby
         /Users/keith/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/console-ring does not exist
[OK]     system
[OK]     moby-syslog
[ERROR]  db
         Connection refused (ECONNREFUSED) connecting to /Users/keith/Library/Containers/com.docker.docker/Data/s40: check if service is running
         com.docker.db is not running
[OK]     env
[OK]     virtualization kern.hv_support
[ERROR]  slirp
         Unexpected error ((Failure
  "Error connecting socket to 9p endpoint unix:/Users/keith/Library/Containers/com.docker.docker/Data/s51: Unix.Unix_error(Unix.ECONNREFUSED, \"connect\", \"\")")) connecting to /Users/keith/Library/Containers/com.docker.docker/Data/s51
         com.docker.slirp is not running
         slirp check failed with: (Failure
  "Error connecting socket to 9p endpoint unix:/Users/keith/Library/Containers/com.docker.docker/Data/s52: Unix.Unix_error(Unix.ECONNREFUSED, \"connect\", \"\")")
[ERROR]  osxfs
         com.docker.osxfs is not running
[OK]     moby-console
[OK]     logs
[ERROR]  docker-cli
         Connection refused (ECONNREFUSED) connecting to /var/run/docker.sock: check if service is running
         Connection refused (ECONNREFUSED) connecting to /Users/keith/Library/Containers/com.docker.docker/Data/s60: check if service is running
         docker ps failed
[OK]     menubar
[OK]     disk

**Comment by djs55** [comment]
Thanks for your report.

Looking in the logs I see
```
2017-02-08 09:01:04+00 ... Docker[com.docker.driver.amd64-linux][1421] <Notice>: mirage_block_open: block_config = file:///Users/keith/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/Docker.qcow2?sync=1&buffered=1 and qcow_config = discard=true;compact_after_unmaps=0
2017-02-08 09:01:04+00 ... Docker[com.docker.driver.amd64-linux][1421] <Notice>: com.docker.hyperkit: [ERROR] Found two references to cluster 25992: 2289.4864 and 98.18944
2017-02-08 09:01:04+00 ... Docker[com.docker.driver.amd64-linux][1421] <Notice>: com.docker.hyperkit: [ERROR] Mirage block device raised exception: (Failure "Found two references to cluster 25992: 2289.4864 and 98.18944")
2017-02-08 09:01:04+00 ... Docker[com.docker.driver.amd64-linux][1421] <Notice>: Mirage-block error: Unknown (Failure "Found two references to cluster 25992: 2289.4864 and 98.18944")
```
This means that the file `~/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/Docker.qcow2` has become corrupt and cannot be safely written to :( The only workaround is to delete this file and relaunch the app -- unfortunately this means that containers and images will need to be rebuilt.

I'm not sure what the root cause is -- the logs are unfortunately not very helpful. Could you tell me whether you opted to import containers from toolbox when you installed? There could be a problem with the import process. Has this happened before? Do you have any ideas how I could reproduce this?

Thanks again for the report, sorry I couldn't be more helpful.

**Comment by avsm** [comment]
@khards just wondering how you are getting along. It would be helpful to us to know the answer to this:
>  Could you tell me whether you opted to import containers from toolbox when you installed? There could be a problem with the import process. Has this happened before? Do you have any ideas how I could reproduce this?

But if you don't have time, we can just close this issue as well.  Thanks for any help!

**Comment by khards** [comment]
Hi, I deleted some files I found mentioned in another bug ticket. (sorry can't recall which ones)
This fixed the issue.
I think the best thing to do would be to trap this error and offer the user a chance to 'repair' or remove these files on startup.

I guess that's the biggest flaw of osx that it does not have an installer / uninstaller or package manager.

**Comment by avsm** [comment]
No problem -- we are continuing to track this issue as part of https://github.com/docker/for-mac/issues/1159, so I will close this for now. Glad that you are unblocked for the moment.  We have also filed another issue about making it easier to repair these files on startup.
